tests/lib/eslint/flat-eslint.js

```diff
@@ -3707,6 +3723,36 @@ describe("FlatESLint", () => {
                 nodePlugin.rules["no-new-require"].meta
             );
         });
+
+        it("should ignore messages not related to a rule", async () => {
+            const engine = new FlatESLint({
+                overrideConfigFile: true,
+                overrideConfig: { rules: { "no-var": "warn" } },
+                reportUnusedDisableDirectives: "warn"
+            });
+
+            const results = [
+                ...await engine.lintText("syntax error"),
+                ...await engine.lintText("// eslint-disable-line no-var"),
+                ...await engine.lintText("", { filePath: "/.ignored.js", warnIgnored: true })
+            ];
+            const rulesMeta = engine.getRulesMetaForResults(results);
```

Comment:
Thanks @mdjermanovic, I see the point here. I've changed the test as you suggested. Now I'm calling `getRulesMetaForResults` once for each array returned by `lintText`. It would be also possible to split the code in three test if that seems clearer, but I haven't done so yet.

BTW, I hope that it will still be possible to use `getRulesMetaForResults` on a merged results array with the
new API, even if that won't be representative of canonical usage (because calling `getRulesMetaForResults` multiple times and merging the returned objects won't be considered better practice, I think).